### PR TITLE
Assignment1 - Nolan Posey & Josh Merrell

### PR DESCRIPTION
**Pull request by Nolan-Posey**
@joshmerrell

### CODE REVIEW

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void GetFilePath_ReturnsFileThatExists()
             Assert.AreEqual(expectedString, percentage);
         }
 
-
+        [TestMethod]
```
Comment (arose-13) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:75]: Not sure this is necessary. This method is used by other test methods to create a file with basic questions, and doesn't actually test any functionality on it's own.

**Review by arose-13 — COMMENTED** [review]
Great job, and nice new feature!

Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
EXTRA CREDIT
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 2]
```diff
@@ -89,8 +92,34 @@ public static Question[] LoadQuestions(string filePath)
                 question.Answers[1] = answer2;
                 question.Answers[2] = answer3;
                 question.CorrectAnswerIndex = correctAnswerIndex;
+                questions[i] = question;
             }
             return questions;
         }
+
+        //Proposed new feature for the app
```
Comment (Bhopss152) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:100]: Here maybe add a method that handles if somehow a null or negative number is thrown.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 3]
```diff
@@ -86,5 +86,17 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
                 File.AppendAllLines(filePath, lines);
             }
         }
+
```
Comment (Bhopss152) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:89]: Adding a comment here to explain the extra feature could be beneficial.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 4]
```diff
@@ -86,5 +86,17 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+        //Test method for suggested new feature
+        [TestMethod]
+        [DataRow(0, "Count Rugen")]
+        [DataRow(5, "Buttercup")]
+        [DataRow(7, "Wesley")]
+        public void GetCharacterName_ReturnsNameOfCharacter(int numberOfCorrectGuesses, string expectedName)
+        {
+            string characterName = Program.GetCharacterName(numberOfCorrectGuesses);
+
+            Assert.AreEqual(characterName, expectedName);
```
Comment (mmwoodfo) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:99]: Good test, typically the expected result is the first parameter, and the actual result is the second parameter. This doesn't change the way the assert works, but it's more so a coding standard that is typically followed in unit tests.

**Pull request closed without merging**